From 2736cc66676b967c0961ca7802e3483b0d8434a0 Mon Sep 17 00:00:00 2001
From: Dominique Leuenberger <dimstar@opensuse.org>
Date: Fri, 24 Feb 2023 13:38:56 +0100
Subject: [PATCH] build: fix awk scripts to work properly when used againts
 sqlite 3.41.0

 The SQL standard requires double-quotes around identifiers and single-quotes around string literals. For example:

    "this is a legal SQL column name"
    'this is an SQL string literal'

With sqlite 3.41.0, this is being enforced on the CLI
---
 data/english.awk | 8 ++++----
 data/table.awk   | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/data/english.awk b/data/english.awk
index c5dc0aa2..8b32f3f2 100644
--- a/data/english.awk
+++ b/data/english.awk
@@ -5,9 +5,9 @@ BEGIN {
     print "BEGIN TRANSACTION;"
 
     # Create english table
-    print "CREATE TABLE IF NOT EXISTS \"english\" ( "   \
-        "\"word\" TEXT NOT NULL PRIMARY KEY,"           \
-        "\"freq\" FLOAT NOT NULL DEFAULT(0)"            \
+    print "CREATE TABLE IF NOT EXISTS 'english' ( "   \
+        "'word' TEXT NOT NULL PRIMARY KEY,"           \
+        "'freq' FLOAT NOT NULL DEFAULT(0)"            \
         ");";
 
     # Create desc table
@@ -16,7 +16,7 @@ BEGIN {
 }
 
     # Insert data into english table
-    {   printf "INSERT INTO english (word, freq) VALUES (\"%s\", %f);\n", $1, $2}
+    {   printf "INSERT INTO english (word, freq) VALUES ('%s', %f);\n", $1, $2}
 
     #quit sqlite3
 END {
diff --git a/data/table.awk b/data/table.awk
index 700529cf..c097ec26 100644
--- a/data/table.awk
+++ b/data/table.awk
@@ -21,7 +21,7 @@ BEGIN {
 
 # Insert data into phrases table
 NF == 4 {
-    printf "INSERT INTO phrases (id, tabkeys, phrase) VALUES (%d, \"%s\", \"%s\");\n", id, $3, $1;
+    printf "INSERT INTO phrases (id, tabkeys, phrase) VALUES (%d, '%s', '%s');\n", id, $3, $1;
     id++;
 }
 
